addons/sale_project/static/src/components/project_right_side_panel/components/project_profitability.js

```diff
@@ -0,0 +1,77 @@
+/** @odoo-module */
+
+import { patch } from "@web/core/utils/patch";
+import { useService } from '@web/core/utils/hooks';
+import { ProjectProfitability } from '@project/components/project_right_side_panel/components/project_profitability';
+import { formatFloat, formatFloatTime } from "@web/views/fields/formatters";
+
+patch(ProjectProfitability.prototype, {
+
+    /*props : {
+        ...ProjectProfitability.props,
+        projectId: Number,
+        context: Object,
+    },*/
```

Comment:
We cannot alter the props inside that patch function, that patch will alter the methods and attribute available in the object created by `ProjectProfitability` class. The `props` is a static attribute meaning "it is a attribute linked to the class and not the object of that class".

To alter the props, we have use the patch on the class and not the prototype of the class or using `Object.assign`. 🙂

```js
patch(ProjectProfitability, {
    props: {
        ...ProjectProtability.props,
        projectId: Number,
        context: Object,
    },
});
```

I think now it is the patch method that we prefer to use but I think those both alternatives also work:

- `Object.assign`

```js
Object.assign(ProjectProfitability.props, {
    projectId: Number,
    context: Object,
});
```

- `reassign new props`

```js
ProjectProfitability.props = {
    ...ProjectProfitability.props,
    projectId: Number,
    context: Object,
};
```